addons/payment_alipay/models/alipay.py

```diff
@@ -0,0 +1,246 @@
+# -*- coding: utf-'8' "-*-"
+
+import base64
+
+try:
+    import simplejson as json
+except ImportError:
+    import json
+import logging
+import urlparse
+import werkzeug.urls
+import urllib2
+
+import util
+from urllib import urlencode, urlopen
+
+from openerp.addons.payment.models.payment_acquirer import ValidationError
+from openerp.addons.payment_alipay.controllers.main import AlipayController
+from openerp.osv import osv, fields
+from openerp.tools.float_utils import float_compare
+from openerp import SUPERUSER_ID, api
+
+import sys
+
+reload(sys)
+sys.setdefaultencoding('utf8')
+
+_logger = logging.getLogger(__name__)
+
+
+class AcquirerAlipay(osv.Model):
+    _inherit = 'payment.acquirer'
+
+    def _get_alipay_urls(self, cr, uid, environment, context=None):
+        """ Alipay URLS """
+        if environment == 'prod':
+            return {
+                'alipay_url': 'https://mapi.alipay.com/gateway.do?'
+            }
+        else:
+            return {
+                'alipay_url': 'https://openapi.alipaydev.com/gateway.do?'
+            }
+
+    def _get_providers(self, cr, uid, context=None):
+        providers = super(AcquirerAlipay, self)._get_providers(cr, uid, context=context)
+        providers.append(['alipay', 'Alipay'])
+        return providers
+
+    @api.one
+    def _get_alipay_partner_key(self):
```

Comment:
you are mixing old and new api, you should choose one and keep it at least for your full class.
anyway, do you need a method just to return this field ?